usr/share/rear/layout/prepare/GNU/Linux/210_load_multipath.sh

```diff
@@ -34,15 +34,59 @@ blacklist {
     fi
 
     LogPrint "Activating multipath"
-    modprobe dm-multipath >&2
-    multipath >&2
-    if [ $? -ne 0 ] ; then
-        LogPrint "Failed to activate multipath, or no multipath device found."
-        rear_shell "Did you activate the multipath devices?"
-    else
-        LogPrint "multipath activated"
+    list_mpath_device=1
+    modprobe dm-multipath >&2 && LogPrint "multipath activated" || LogPrint "Failed to load dm-multipath module"
+
+    # Asking to the User what to do after multipath command return 1.
+    # It could be because no multipath device were found (sles11/rhel6)
+    # or a real problem in the multipath configuration.
+    prompt="Failed to get multipath device list or no multipath device found."
+
+    rear_workflow="rear $WORKFLOW"
+    unset choices
+    choices[0]="Multipath is not needed. Continue recovery."
+    choices[1]="Run multipath with debug options."
+    choices[2]="Enter into rear-shell to manually debug multipath."
+    choices[3]="Abort '$rear_workflow'"
+    choice=""
+    wilful_input=""
+
+    # looping on this menu while multipath failed to list device.
+    # Note: On sles11/rhel6, multipath failed if no multipath device is found.
+    while ! multipath ; do
+        echo
+        choice="$( UserInput -p "$prompt" -D "${choices[0]}" "${choices[@]}")"&& wilful_input="yes" || wilful_input="no"
+        case "$choice" in
+            (${choices[0]})
+                # continue recovery without multipath
+                is_true "$wilful_input" && LogPrint "User confirmed continuing without multipath" || LogPrint "Continuing '$rear_workflow' by default"
+                LogPrint "You should consider removing BOOT_ON_SAN parameter from your rear configuration file if you don't need multipath on this server."
+
+                # Avoid to list mpath device.
+                list_mpath_device=0
+                break
+                ;;
+            (${choices[1]})
+                # Run multipath in debug level
+                LogPrint "starting multipath -v3 (debug mode)"
+                multipath -v3
+                ;;
+            (${choices[2]})
+                # Exit to shell to debug multipathing issue
+                rear_shell "Do you want to go back to '$rear_workflow' ?"
+                ;;
+            (${choices[3]})
+                # Abort rear recovery
+                abort_recreate
+                Error "User chose to abort '$rear_workflow' in ${BASH_SOURCE[0]}"
+                ;;
+        esac
+    done
+
+    # Search and list mpath device.
+    if is_true $list_mpath_device ; then
+        LogPrint "Listing multipath device found"
         dmsetup ls --target multipath
```

Comment:
LogPrint shows only STDOUT to the user
but has both STDOUT and STDERR in the log:
<pre>
LogPrint "$( cat qqq /etc/os-release QQQ )"
</pre>
at the beginning of init/default/030_update_recovery_system.sh
results on the terminal (in verbose mode):
<pre>
e205:~/rear.master # usr/sbin/rear -v mkrescue
Relax-and-Recover 2.2 / Git
Using log file: /root/rear.master/var/log/rear/rear-e205.log
NAME="SLES"
VERSION="12-SP2"
VERSION_ID="12.2"
PRETTY_NAME="SUSE Linux Enterprise Server 12 SP2"
ID="sles"
ANSI_COLOR="0;32"
CPE_NAME="cpe:/o:suse:sles:12:sp2"
Creating disk layout
...
</pre>
and in te log there is
<pre>
2017-08-24 13:29:25.909871995 Including init/default/030_update_recovery_system.sh
cat: qqq: No such file or directory
cat: QQQ: No such file or directory
2017-08-24 13:29:25.917933198 NAME="SLES"
VERSION="12-SP2"
VERSION_ID="12.2"
PRETTY_NAME="SUSE Linux Enterprise Server 12 SP2"
ID="sles"
ANSI_COLOR="0;32"
CPE_NAME="cpe:/o:suse:sles:12:sp2"
2017-08-24 13:29:25.925857041 Including init/default/050_check_rear_recover_mode.sh
</pre>

Currently there is no function to get
both STDOUT and STDERR shown to the user
plus logging them both.

I think I will add one...